### ISSUE TITLE
Not working?

### ISSUE THREAD

**Issue by paaljoachim** [bueltge/Multisite-Global-Media#1]
Hey

I installed and activated your plugin.
Added a few images to various sites.
Went to a post in one of the sites. Added media -> clicked Global Media and the area is blank.
I did expect to find all the other sites media when clicking on Global Media (All media items drop down is selected).

What am I supposed to do to have the various media from the various sites show?

I am using a local development site built in MAMP.

**Comment by bueltge** [comment]
The blog ID is exist? Inside the source do you find the constant for Tue blog ID of the site, there store all media for all other blogs.

**Comment by paaljoachim** [comment]
How do I check this?
Could you do a step for step procedure?

**Comment by bueltge** [comment]
This constant store the id o this blog, site there store all media files, there is the primary blog, there share all file for the other blogs in the network. On default you can't see this id in the back end. But with the plugin Multisite Enhancement or a mouse over about the edit site link see you the id of each blog. I use often the second blog as central media store for share to all other blogs. All media files from this blog will share about the new Global Media view.

**Comment by paaljoachim** [comment]
Ok.
Constant stores the id of a blog. The ID says 3 meaning as I see it blog 3.
One has the primary blog of 1 and various other blogs that have their own id's. Check.

I have the Multisite Enhancement plugin installed and can see the ID's i the All Sites screen. Check.

You said: I use often the second blog (meaning the blog with ID of 3) as a central media store to share media with all other blogs.

How do I select a blog to use as the central place to store media? 
How will this blog share its media items with all other blogs?

What about adding a media library sidebar item to the Network Admin sidebar? Creating a Network Admin media library to where one can choose to upload media that will automatically be shared with all the blogs/sites?

Bottom line. It would be nice to on any blog/site -> go to a post or page -> click Add Media -> click Global Media and see all media added to the Network. 

Btw the All Media drop down is very tiny and could be made wider by changing max-width from 47% to 87%.

**Comment by bueltge** [comment]
>You said: I use often the second blog (meaning the blog with ID of 3) as a central media store to share media with all other blogs.

I mean the blog with ID 2. The first, the 1, is my administrator blog, often with a lot of enhancements to see information about all blogs, users and much more. This is my store for the administration.
The blog 2 is the store for the medias, there will share about all blogs.

>How do I select a blog to use as the central place to store media? 
The set of the constant for the blog id inside the blog define the blog, there store the media and share this. The idea is, that we not enhance a lot of custom source. We use simple a blog id and use the default WordPress functionalities. Also give it this the chance, that media administrators control the media  uploads. Often useful for share media, in the law topic.

>How will this blog share its media items with all other blogs?
The plugin add the button "Global Media" in all media modal of all other blogs inside the network and display all media files from the Media Blog, like ID 2 from my example. 

What about adding a media library sidebar item to the Network Admin sidebar? Creating a Network Admin media library to where one can choose to upload media that will automatically be shared with all the blogs/sites?
This is a good idea. Currently we have only a small plugin to share media, not more. I think the plugin and his idea have a lot of improvements for a helpful plugin, like upload from each blog to the global media blog. Currently we have a solution inside this plugin to  add media files to the global media. 

> Btw the All Media drop down is very tiny and could be made wider by changing max-width from 47% to 87%.
On my tests was is in the same size as all other media dialogs. See the screenshot from my tests -- https://github.com/bueltge/Multisite-Global-Media/blob/master/assets/screenshot-1.png

Best regards and welcome for more feedback and also pull requests.
Frank

**Comment by paaljoachim** [comment]
Ahhhh.... I finally understand.

Btw I would suggest adding the following to the readme:

To get Global Media to work one has to follow these steps:
1. Decide on which blog/site that will host shared media for the network.
2. Add media to the media library for the specific blog/site.
4. Find the ID of a site by going to All Sites section hovering over the site checking the status line bottom left or installing a plugin that shows the ID's of each site.
3. Open the file multisite-global-media.php go to the following code:

/**
 * Select the ID of the site/blog to where you want media that will be shared across the network to be stored. (---I rephrased making it clear what is needed to get this to work---).
 *
 * @var    integer
 * @since  2015-01-22
 */

const BLOG_ID = 3;

----

You said:  "Currently we have a solution inside this plugin to add media files to the global media."
How? Have you added this to the current plugin?

---

Add Media -> Click Global Media -> check out the drop down.
Notice only the words All Med are seen.

![screen shot 2015-02-27 at 13 03 18](https://cloud.githubusercontent.com/assets/5323259/6412195/0d870590-be81-11e4-8b30-a0b9ae2bbf75.png)

---

**Comment by bueltge** [comment]
Thanks a lot, I had update the readme with your hints.

> We have not

Sorry, the not was forgotten from my side.

The width if the select field is now changed, new width with custom style. Please download and test again. Thanks.

**Comment by paaljoachim** [comment]
4. Find the ID of a site by going to All Sites section hovering over the site checking the left bottom status bar left or installing a plugin that shows the ID's of each site.

I had two left words in there..:) Remove the second word left...

I would also suggest going through the language rephrasing for the rest. Nice plugin btw!

**Comment by bueltge** [comment]
Thanks again.
If you have more improvements for my bad English, please feel free to write a better readme!

**Comment by paaljoachim** [comment]
I wrote a lot of changes in the post a few posts above here where I also posted the screenshot. Go through it.... I am also going to add the adjustments to the readme and text in the php file below here.

Multisite Global Media

Multisite Global Media is a WordPress plugin which shares media across the Multisite network.

Description

This small plugin adds a new tab to the media modal which gives you the opportunity to share media from one site to all the other sites of the network. The multisite-global-media.php file uses the ID of the site that will store the global media. Currently the site id is set at const BLOG_ID = 3. Change this value to set one of the other sites as the default for storing global media.

NB! Do also change the text in the php file 
Select the ID of the site/blog to where you want media that will be shared across the network to be stored. - Description in the php file regarding using the side ID.

**Comment by bueltge** [comment]
Thanks again!
I have change all your ideas, rephrasing for a better understanding of the plugin.
